### PR TITLE
refactor: class APIError should extend Error interface

### PR DESCRIPTION
**Pull request by sherwinski**
This PR updates the `APIError` class to be more idiomatic and extend the `Error` interface. It also expands the TS tests to account for Promises returned by the `request()` method.

Relates to #67

### CODE REVIEW

File: types/imgix-api-test.ts [hunk 1]
```diff
@@ -25,7 +26,24 @@ ix.request(`sources/upload/${SOURCE_ID}/image.jpg`, {
...
 // $ExpectType Promise<RequestResponse | RequestError>
-ix.request(`sources`, {
+ix.request('sources', {
   method: 'POST',
   body: BODY_JSON,
 });
+
+// $ExpectType Promise<void>
+ix.request('sources').then((response) => {
+  response = response as RequestResponse;
```
Comment (sherwinski) [inline types/imgix-api-test.ts:36]: This one's kind of iffy and might as well be excluded. Curious if there are any suggestions on how to do this better.

File: src/api-error.js [hunk 2]
```diff
@@ -1,23 +1,27 @@
...
+    this.message = message;
+    this.status = status;
+  }
 
-  this.toString = () =>
-    `${this.message}\nResponse:\n${
-      isObject ? JSON.stringify(this.response, null, 2) : this.response
+  toString = () => {
+    return `\n${this.message}\nResponse:\n${
+      this.isObject ?  this.response : JSON.stringify(this.response, null, 2)
```
Comment (frederickfogerty) [inline src/api-error.js:22]: Was this switched around accidentally?

File: types/imgix-api-test.ts (hunk 1 above)
Comment (frederickfogerty) [reply types/imgix-api-test.ts:36]: I think this is okay, but I would just drop line 34 since it's not actually testing anything

**Review by frederickfogerty — COMMENTED** [review]
LGTM apart from the one switcharound mistake (I think)

File: src/api-error.js (hunk 2 above)
Comment (sherwinski) [reply src/api-error.js:22]: The switch was intentional but on second look I now realize it wasn't the best way to go about fixing an issue. I'll open a separate PR to explore the issue.

Comment (sherwinski) [reply src/api-error.js:22]: Address in #75

**Pull request merged**